packages/react/src/components/Storage/FileUploader/FileUploader/FileUploader.tsx

```diff
@@ -41,6 +43,13 @@ export function FileUploader({
     );
   }
 
+  const displayText = useMemo(() => {
+    return {
+      ...defaultFileUploaderDisplayText,
+      ...overrideDisplayText,
+    };
+  }, [overrideDisplayText]);
```

Comment:
Good point, let me test that use-case

Reply:
I did a quick example of toggling strings and it works, although the `useMemo` here is unnecessary because changing the `displayText` prop will always cause a re-render and will always run the memo function right? 

```jsx
import { Button, FileUploader } from '@aws-amplify/ui-react'; // IGNORE
import { useState } from 'react';

const es = {
  dropFilesText: 'yoink filez',
  uploadButtonText: (count) => `YOINK ${count} filez`
}

const en = {
  dropFilesText: 'yeet filez',
  uploadButtonText: (count) => `yeet ${count} filez`
}

export const DefaultFileUploaderExample = () => {
  const [text, setText] = useState(en);
  
  return (
    <>
    <Button onClick={() => {setText(en)}}>En</Button>
    <Button onClick={() => {setText(es)}}>Es</Button>
    <FileUploader
      acceptedFileTypes={['image/*']}
      accessLevel="public"
      displayText={text}
      provider="fast" // IGNORE
    />
    </>
  );
};

```

Reply:
Changing the `displayText` prop in your example would cause the useMemo to rerun because the objects, while static, ultimately are different object instances. 

We probably don't need to memoize here imo, in particular if we update the `FileUploader` internals to evaluate all the display text in the parent (this file) we would just be passing strings which do not benefit from memoization

Reply:
Makes sense, removed the useMemo here